apps/admin_app/lib/admin_app_web/controllers/product_controller.ex

```diff
@@ -112,6 +112,15 @@ defmodule AdminAppWeb.ProductController do
     redirect_with_updated_conn(conn, params)
   end
 
+  def toggle_variant_state(conn, %{"state" => state, "id" => product_id}) do
```

Comment:
We are getting state from frontend it might be possible that the state can be invalid.
Porduct update will fail with changeset. That clause needs to be handled.